src/dialects/ibmi/data-types.js

```diff
@@ -0,0 +1,331 @@
+    }
+
+    static parse(value) {
+      if (value) {
+        return true;
+      }
+
+      return false;
+    }
+  }
+
+  class STRING extends BaseTypes.STRING {
+    toSql() {
+      if (this._binary) {
+        return `BINARY(${this._length})`;
+      }
+
+      return super.toSql(this);
+    }
+
+    _stringify(value, options) {
+      if (Buffer.isBuffer(value)) {
+        return `BLOB(X'${value.toString('hex')}')`;
+      }
+
+      return value.toString();
+    }
+
+  }
+
+  class TEXT extends BaseTypes.TEXT {
+    toSql() {
+      let len = 0;
+      if (this._length) {
+        if (isNaN(this._length)) {
+          if (typeof this._length === 'string') {
+            switch (this._length.toLowerCase()) {
+              case 'tiny':
+                this._length = 256; // tiny = 2^8
+                break;
+              case 'medium':
+                this._length = 8192; // medium = 2^13 = 8k
+                break;
+              case 'long':
+                this._length = 65_536; // long = 64k
+                break;
+            }
+          } else {
+            this._length = 8192;
+          }
+        }
+
+        if (len > 0) {
+          this._length = len;
+        }
+      } else {
+        this._length = 8192;
+      }
+
+      if (this._length > 32_672) {
+        len = `CLOB(${this._length})`;
+      } else {
+        len = `VARCHAR(${this._length})`;
+      }
+
+      return len;
+    }
+  }
+
+  class CHAR extends BaseTypes.CHAR {
+    toSql() {
+      if (this._binary) {
+        return `CLOB(${this._length})`;
+      }
+
+      return super.toSql();
+    }
+  }
+
+  class NUMBER extends BaseTypes.NUMBER {
+    toSql() {
+      let result = this.key;
+      if (this._length) {
+        result += `(${this._length}`;
+        if (typeof this._decimals === 'number') {
+          result += `,${this._decimals}`;
+        }
+
+        result += ')';
+      }
+
+      return result;
+    }
+  }
+
+  class SMALLINT extends BaseTypes.SMALLINT {
+    constructor(length) {
+      super(length);
+      removeUnsupportedIntegerOptions(this);
+    }
+  }
+
+  class INTEGER extends BaseTypes.INTEGER {
+    constructor(length) {
+      super(length);
+      removeUnsupportedIntegerOptions(this);
+    }
+  }
+
+  class BIGINT extends BaseTypes.BIGINT {
+    constructor(length) {
+      super(length);
+      removeUnsupportedIntegerOptions(this);
+    }
+
+    _stringify(value) {
+      return value.toString();
+    }
+
+    static parse(value) {
+      return value.toString();
+    }
+  }
+
+  class FLOAT extends BaseTypes.FLOAT {
+    constructor(length, decimals) {
+      super(length, decimals);
+      if (this._decimals) {
+        warn('Db2 for i does not support FLOAT with decimal scale. FLOAT will be used instead. If you want to specify scale, use DECIMAL or NUMERIC data types.');
+        this._length = undefined;
+        this._decimals = undefined;
+      }
+    }
+  }
+
+  class DOUBLE extends BaseTypes.DOUBLE {}
```

Comment:
Haven't checked with the other dialects, but can this just be empty?

Reply:
It will inherit everything, constructors included.
However, is it something we need to do? Or can this line just be removed?